server/internal/server/models.go

```diff
@@ -57,22 +57,47 @@ func (s *S) ListModels(
 		return nil, fmt.Errorf("failed to extract user info from context")
 	}
 
-	var modelProtos []*v1.Model
-	// First include base models.
-	bms, err := s.store.ListBaseModels(userInfo.TenantID)
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "list models: %s", err)
+	if req.Limit < 0 {
+		return nil, status.Errorf(codes.InvalidArgument, "limit must be non-negative")
 	}
-	for _, m := range bms {
-		if !isBaseModelLoaded(m) && !req.IncludeLoadingModels {
-			continue
+	limit := req.Limit
+	if limit == 0 {
+		limit = defaultPageSize
+	}
+	if limit > maxPageSize {
+		limit = maxPageSize
+	}
+
+	var modelProtos []*v1.Model
+	// First include base models. Base models is not paginated.
```

Comment:
Could you add a TODO comment to paginate base models as well? The frontend code will just show the list of items returned from the response.